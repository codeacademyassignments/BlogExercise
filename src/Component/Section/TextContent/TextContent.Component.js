import React from 'react';
import '../../index.css';
import PropTypes from 'prop-types';
import ArtFooter from '../ArtFooter/ArtFooter.component';

const TextSection = ({ textContent, onClapClick, onHeartClick }) => (
  <section className="desc">
    <span className="left-caption">{textContent.date}</span>
    <span className="right-caption">{textContent.readingTime}</span>
    <br />
    <p className="title">
      {textContent.title}
      {' '}
    </p>
    <p className="description">{textContent.description}</p>
    <hr />
    <ArtFooter liked={textContent.liked} claps={textContent.claps} onClapClick={onClapClick} onHeartClick={onHeartClick} />
  </section>
);

TextSection.propTypes = {
  onClapClick: PropTypes.func.isRequired,
  onHeartClick: PropTypes.func.isRequired,
  textContent: PropTypes.shape({
    date: PropTypes.string.isRequired,
    readingTime: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,

};
export default TextSection;
