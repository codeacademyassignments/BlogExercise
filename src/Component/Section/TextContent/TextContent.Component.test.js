import React from 'react';
import renderer from 'react-test-renderer';
import TextContent from './TextContent.Component';
import {shallow} from 'enzyme';

const blog = {
  "date": "2nd Januray, 2018",
  "readingTime": "2 mins",
  "title": "The future of abstract art and the culture ...",
  "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
  "claps": 10,
  "liked": false,
}
describe('snapshot testing',()=>{
  it('snapshot should match',()=>{
    const tree = renderer.create(<TextContent textContent={blog} onClapClick={jest.fn()} onHeartClick={jest.fn()}/>);
    expect(tree).toMatchSnapshot();
  })
})

describe('ArtFooter',()=>{
  const mockOnClapClick = jest.fn();
  const mockOnHeartClick = jest.fn();
  const wrapper = shallow(<TextContent textContent={blog} onClapClick={mockOnClapClick} onHeartClick={mockOnHeartClick}/>);
  it('should have the mockOnClapClick function as onClapClick props',()=>{
    expect(wrapper.find('ArtFooter').props().onClapClick).toEqual(mockOnClapClick);
  })
  it('should have the mockOnHeartClick function as onHeartClick props',()=>{
    expect(wrapper.find('ArtFooter').props().onHeartClick).toEqual(mockOnHeartClick);
  })
})