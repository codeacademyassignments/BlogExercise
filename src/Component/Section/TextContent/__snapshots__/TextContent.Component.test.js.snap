// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContent snapshot should match 1`] = `
<section
  className="desc"
>
  <span
    className="left-caption"
  >
    2nd Januray, 2018
  </span>
  <span
    className="right-caption"
  >
    2 mins
  </span>
  <br />
  <p
    className="title"
  >
    The future of abstract art and the culture ...
     
  </p>
  <p
    className="description"
  >
    Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
  </p>
  <hr />
  <img
    alt="clapping"
    className="claps"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "clapping.svg",
      }
    }
  />
  <span
    className="no-of-claps"
  >
    10
  </span>
  <img
    alt="heart-black"
    className="heart"
    onClick={[Function]}
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "heart-black.svg",
      }
    }
  />
</section>
`;
